internal/guest/commonutils/utilities.go

```diff
@@ -24,3 +33,72 @@ func DecodeJSONWithHresult(r io.Reader, v interface{}) error {
 	}
 	return nil
 }
+
+// StartTimeSyncService starts the `chronyd` deamon to keep the UVM time synchronized.  We
+// use a PTP device provided by the hypervisor as a source of correct time (instead of
+// using a network server). We need to create a configuration file that configures chronyd
+// to use the PTP device.  The system can have multiple PTP devices so we identify the
+// correct PTP device by verifying that the `clock_name` of that device is `hyperv`.
+func StartTimeSyncService() error {
+	ptpClassDir, err := os.Open("/sys/class/ptp")
+	if err != nil {
+		return errors.Wrap(err, "failed to open PTP class directory")
+	}
+
+	ptpDirList, err := ptpClassDir.Readdirnames(-1)
+	if err != nil {
+		return errors.Wrap(err, "failed to list PTP class directory")
+	}
+
+	var ptpDirPath string
+	found := false
+	expectedClockName := "hyperv"
+	for _, ptpDirPath = range ptpDirList {
+		clockNameFilePath := filepath.Join(ptpClassDir.Name(), ptpDirPath, "clock_name")
+		clockNameFile, err := os.Open(clockNameFilePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to open clock name file at %s", clockNameFilePath)
+		}
+		// Expected clock name is `hyperv` so read first 6 chars and verify the
+		// name
+		buf := make([]byte, len(expectedClockName))
+		fileReader := bufio.NewReader(clockNameFile)
+		_, err = fileReader.Read(buf)
+		if err != nil {
+			return errors.Wrapf(err, "read file %s failed", clockNameFilePath)
+		}
+		clockName := string(buf)
+		if strings.EqualFold(clockName, expectedClockName) {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return errors.Errorf("no PTP device found with name \"%s\"", expectedClockName)
+	}
+
+	// create chronyd config file
+	ptpDevPath := filepath.Join("/dev", filepath.Base(ptpDirPath))
+	chronydConfigString := fmt.Sprintf("refclock PHC %s poll 3 dpoll -2 offset 0\n", ptpDevPath)
+
+	chronydConfPath := "/tmp/chronyd.conf"
```

Comment:
Might as well have this as a global const

Reply:
That can hurt readability. If you're reading through code and see a global, you suddenly have to do extra work to figure out where it's being used. If it's only needed in this function then I would just keep it scoped to this function, though it could be moved to a `const` block in this function.

Reply:
const in this function is fine with me. Global const in the communutils package might be a bit odd for this anyways, but truthfully maybe this entire function doesn't belong in communtils looking again

Reply:
Oh you had the same feedback here haha https://github.com/microsoft/hcsshim/pull/1119#discussion_r691566466